src/auth.ts

```diff
@@ -52,11 +53,30 @@ export const drive = google.drive({
  * Requests authorization to manage Apps Script projects.
  * @param {boolean} useLocalhost True if a local HTTP server should be run
  *     to handle the auth response. False if manual entry used.
+ * @param {any} creds either boolean or string. If true (boolean), then --creds
+ *     was used without any input file. Otherwise creds is a string of the file
+ *     they entered.
  */
-async function authorize(useLocalhost: boolean, writeToOwnKey: boolean) {
+async function authorize(useLocalhost: boolean, creds: any) {
```

Comment:
Can we never have any?
`string|boolean`?

Reply:
Typescript is not a fan 😔

```
[ts]
Argument of type 'string | false' is not assignable to parameter of type 'string | number | Buffer | URL'.
  Type 'false' is not assignable to type 'string | number | Buffer | URL'.
```

when trying to read the credentials file:

```
    const credentials = JSON.parse(fs.readFileSync(creds, 'utf8'));
```

Reply:
The main reason for doing this is this line:

```
  creds = creds === true ? 'credentials.json' : creds;
```

and the reason for that was because if you put a default value in commander, it sets `options.creds` to that value whether or not the user entered in the `--creds` flag.

So I removed the default value from the commander side of things and use this line to set it to `credentials.json` by default.

Reply:
Can you just do `string` and default `creds` outside the body (in the caller)?

There has to be some way to not use `any`.
This is an internal method too, so we have flexibility to our method signature.